functions-and-scope.js

```diff
@@ -15,38 +23,119 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
+
+console.log("opdracht 1c functie geschikt voor meerdere arrays met elk hun eigen naaam")
+
+//grades is al eerder gedefinieerd
+studentCumLaude = 0; //let is niet meer nodig want al eerder gedeclareerd
+function cumLade (gradeArray) {
+    for (let participants = 0; participants < gradeArray.length; participants++) {
+        if (gradeArray[participants] > 7) {
+            studentCumLaude = studentCumLaude + 1
+        }
+    }
+    return studentCumLaude;
+}
+cumLade(grades);
+console.log("Voor de array grades: " + studentCumLaude);
+
+studentCumLaude = 0;
+cumLade(grades2);
+console.log("Voor de array grades2: " + studentCumLaude);
+
+
 // ---- Verwachte uitkomsten:
-// cumLaude(grades) geeft 6
-// cumLaude([6, 4, 5]) geeft 0
-// cumLaude([8, 9, 4, 6, 10]) geeft 3
+// studentCumLaude(grades) geeft 6
+// studentCumLaude([6, 4, 5]) geeft 0
+// studentCumLaude([8, 9, 4, 6, 10]) geeft 3
 
 
 
 
-/* Opdracht  2: Gemiddeld cijfer */
+/* Opdracht  2: Gemddeld cijfer */
 
 /* 2a: Script schrijven  */
 // De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
 // Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
-// * Hoe wordt een gemiddelde berekend?
-// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
-// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+// * Hoe wordt een gemiddelde berekend? // de waarde van alle entries optellen en delen door het aantal entries
+// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen? // de som en de lengte van de array
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten? for loop van 0 tot array.length-1
 // Log het antwoord in de terminal.
 
+console.log("opdracht 2, gemiddelden")
+//declareer de variabele sumOfGrades
+let sumOfGrades = 0;
+//for loop die door de array itereert
+for(let i =0; i<grades.length; i++) {
+    //alle arraywaardes optellen
+    sumOfGrades += grades[i];
+}
+//gemiddelde uitrekenen
+const averageGrade = sumOfGrades/grades.length;
+//gemiddelde loggen in de console
+console.log(averageGrade);
+
 // ---- Verwachte uitkomst: 6.642857142857143
 
 
 /* 2b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
+console.log("opdracht 2b gemiddelden in een herbruikbare functie")
+function classAverage () {
```

Comment:
Als de functie met andere arrays om moet kunnen gaan, dan moet je een array kunnen meegeven aan deze functie in de vorm van een parameter: `function classAverage(gradesArray) {`